File: client/src/types/review.ts
export interface Review {
  id: string
  productId: string
  userId: string
  userName: string
  rating: number
  comment: string
  createdAt: string
}

export interface CreateReviewRequest {
  rating: number
  comment: string
}

export interface UpdateReviewRequest {
  rating: number
  comment: string
}

export interface PagedResult<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

File: src/Domain/Entities/Review.cs
using ShopeeClone.Domain.Common;

namespace ShopeeClone.Domain.Entities;

public class Review : BaseEntity
{
    public Guid ProductId { get; set; }
    public Product? Product { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string UserName { get; set; } = string.Empty;
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
}
